### ISSUE TITLE
Feature request: Support for gitlab.com

### ISSUE THREAD

**Issue by holbue** [FauDroids/MrHyde#48]
Hi, would it be hard to add also gitlab as git provide?

**Comment by Maddoc42** [comment]
Definitely doable! 👍 

To confirm, you are referring to repositories on gitlab.com and not the self hosted version? For self hosted providers access via SSH keys probably makes more sense.

Adding this to the upcoming release for now, as no major complications are expected. I'll investigate if this is really the case.

**Comment by Maddoc42** [comment]
Unfortunately this doesn't seem to work the same as with GitHub / Bitbucket :open_mouth: GitLab does have an API for getting various info about repositories / projects (https://docs.gitlab.com/ce/api/), but Oauth does not seem to give access to cloning repositories :see_no_evil: 

GitLab supports three types of authentication:
- **Oauth**: gives access to the API, no cloning
- **Personal access tokens**: need to be created by the user in their account, grant access to API + repositories
- **Private tokens**: not sure what those are for, but also don't have the user comfort of Oauth

Long story short, I believe personal access tokens are the only solution here. @holbue are you using any other applications that integrate with GitLab?

**Comment by holbue** [comment]
Oh, what a pity. No, I don't use another app that's integrated with gitlab.com<http://gitlab.com>. And yes, I also use access tokens for my daily work with my repos there. So indeed, it might not be possible with oauth only... :-/

Mit freundlichen Gr??en / Best regards
Holger B?ch (CI/BDT-EU)

Am 09.10.2016 um 23:34 schrieb Philipp Eichhorn <notifications@github.com<mailto:notifications@github.com>>:


Unfortunately this doesn't seem to work the same as with GitHub / Bitbucket ? GitLab does have an API for getting various info about repositories / projects (https://docs.gitlab.com/ce/api/), but Oauth does not seem to give access to cloning repositories ?

GitLab supports three types of authentication:

  *   Oauth: gives access to the API, no cloning
  *   Personal access tokens: need to be created by the user in their account, grant access to API + repositories
  *   Private tokens: not sure what those are for, but also don't have the user comfort of Oauth

Long story short, I believe personal access tokens are the only solution here. @holbue<https://github.com/holbue> are you using any other applications that integrate with GitLab?

-

**Comment by Maddoc42** [comment]
Just released a new version in the beta channel which includes GitLab via the personal access token :dancing_women: let me know if this works for you! :+1:

**Comment by holbue** [comment]
Works perfect! Thank you so much! ☺

Mit freundlichen Grüßen / Best regards

Holger Buech

Application Digital Web Capabilities and Platforms - Technology Web Integration Europe (CI/BDT-EU)
Tel. +49(711)811-0 | Fax +49(711)811 | Holger.Buech@de.bosch.com<mailto:Holger.Buech@de.bosch.com>


Von: Philipp Eichhorn [mailto:notifications@github.com]
Gesendet: Montag, 10. Oktober 2016 18:43
An: FauDroids/MrHyde <MrHyde@noreply.github.com>
Cc: Buech Holger (CI/BDT-EU) <Holger.Buech@de.bosch.com>; Mention <mention@noreply.github.com>
Betreff: Re: [FauDroids/MrHyde] Feature request: Support for gitlab.com (#48)


Just released a new version in the beta channel which includes GitLab via the personal access token 👯 let me know if this works for you! 👍

**Comment by holbue** [comment]
Closed, imho
